crates/bevy_ui/src/layout/ui_surface.rs

```diff
@@ -21,9 +21,26 @@ pub struct RootNodePair {
     pub(super) user_root_node: taffy::NodeId,
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct LayoutNode {
+    // Implicit "viewport" node if this `LayoutNode` corresponds to a root UI node entity
+    pub(super) viewport_id: Option<taffy::NodeId>,
```

Comment:
adding it here means that it will allocate memory for each entry even though only few of them will be actually filled

Reply:
that's an additional `u64` for every UI entity. not sure if it's a problem or not

Reply:
We have really fat UI nodes already so it shouldn't make much difference. I really didn't want to add any more complexity to these modules in this PR and this was the solution I came up with that required the fewest changes to the exisiting code.